README.md

```diff
@@ -1 +1,110 @@
-# metrics-k8s-proxy
\ No newline at end of file
+# metrics-k8s-proxy
+
+**Metrics K8s Proxy** is a lightweight proxy designed to expose a unified metrics endpoint for multiple Kubernetes pods. The proxy watches for pods in a Kubernetes cluster and listens on port `15090`, where it exposes aggregated metrics on a `/metrics` endpoint. 
+
+The proxy itself doesn't scrape metrics. Instead, when Prometheus sends a scrape request, the proxy fans out the request to all registered endpoints that have Prometheus Kubernetes service discovery annotations. It then aggregates the results and appends metadata such as pod names and namespaces, allowing Prometheus to efficiently retrieve metrics from multiple pods.
```

Comment:
If I understand things correctly, we do something like:

* listen on `/metrics`
* when handling `GET` on `/metrics`, trigger a scrape of all watched pods
* collect all metrics from watched pods, annotate them, and send them as a response to `/metrics`

"The proxy itself doesn't scrape metrics" feels incorrect for this workflow.  This program does scrape the metrics, its just that it doesn't *proactively scrape and cache them* but instead reactively scrapes and returns them when someone hits its `/metrics` endpoint.  Maybe we can reword this a little?